# DESeq2_Argiope_bruennichi
Analyzing Argiope_bruennichi RNA-seq data with DESeq2, complete matrix from Kallisto
Analyzing Argiope_bruennichi RNA-seq Data with DESeq2: Complete Matrix from Kallisto

## Overview
This repository contains scripts and resources for analyzing RNA-seq data from Argiope bruennichi using the DESeq2 package in R. The analysis utilizes count data generated by Kallisto, a tool for estimating transcript abundances through pseudoalignment, providing a foundation for differential expression analysis.

Input Data
Data Format:
DESeq2 requires a matrix of integer values representing raw, un-normalized count data. Each row corresponds to a gene, and each column represents a sample.

Un-normalized Counts:
It is crucial to input raw counts into DESeq2, as its statistical model (Love, Huber, and Anders 2014) assesses measurement precision based on these un-normalized values. These counts must accurately portray observed reads or fragments for each gene in each sample.

Library Size Correction:
DESeq2 automatically accounts for library size differences across samples, ensuring that sequencing depth variations do not bias the analysis. Therefore, input counts should remain un-scaled.

Data Set Description
The analysis is based on 96 libraries, including tissues from six individuals, divided across two parts (distal/proximal), with four technical replicates per sample and two ends.

Transcript Quantification with Kallisto
Kallisto was employed to estimate transcript abundances for each sample using a pseudoalignment approach. The output matrix from Kallisto consists of estimated transcript abundances presented as transcripts per million (TPM), with samples as columns and transcripts (or genes) as rows.

## Required Libraries

GenomicFeatures: For working with genomic annotation.
ggplot2: For data visualization.
reshape: For restructuring and aggregating data.
dplyr: For data manipulation.
tidyverse: A collection of R packages for data manipulation and visualization.
DESeq2: For differential expression analysis.
tximport: To import transcript abundance data from Kallisto.
tximportData: Provides example datasets for tximport.
apeglm: For log fold change shrinkage.
pheatmap: For creating heatmaps.
