---
title: "Gene_expression_level_analysis_complete_Matrix"
format: html
editor: visual
author: Zaide Montes-Ortiz, Pheromone Group, Lund University, 2023

---

# Analyzing Argiope_bruennichi RNA-seq data with DESeq2, complete matrix from Kallisto

## Input data

**Input data format:** DESeq2 expects count data in the form of a matrix of integer values. Each element of the matrix represents the number of reads or fragments assigned to a specific gene (rows) in a particular sample (columns). This applies to various high-throughput sequencing experiments like RNA-seq, ChIP-Seq, or quantitative mass spectrometry.

**Un-normalized counts:** The values in the count matrix should be un-normalized counts, which are the raw counts of sequencing reads or fragments. Un-normalized counts accurately reflect the number of reads or fragments observed for each gene in each sample.

**Importance of un-normalized counts:** DESeq2's statistical model (Love, Huber, and Anders 2014) requires the use of un-normalized counts to correctly assess the measurement precision. By utilizing un-normalized counts, DESeq2 can accurately account for the variability and noise present in the raw sequencing data during the differential expression analysis.

**Library size correction:** DESeq2 internally corrects for library size differences between samples, ensuring that the analysis considers the impact of sequencing depth. Therefore, there is no need to provide transformed or normalized values such as counts scaled by library size as input. DESeq2 takes care of library size normalization within its modeling framework.

**Description of data set**

A total of 96 libraries were analyzed, including tissues from six individuals, two parts (distal/proximal), four technical replicates, and two ends.

**Kallisto** was used to estimate the transcript abundances for each sample. Kallisto is a transcript quantification tool that utilizes a pseudoalignment approach to quickly and accurately estimate transcript abundances from RNA-seq data.

The abundance matrix generated by Kallisto represents the estimated transcript abundances for each sample. The matrix will have **the samples as columns** and the **transcripts (or genes) as rows**. The values in the matrix represent the estimated number of transcripts (in transcripts per million, **TPM**) for each sample. \## Required libraries

```{r}
library(GenomicFeatures)
library(ggplot2)
library(reshape)
library(dplyr)
library(tidyverse)
library(DESeq2)
library(tximport)
library(tximportData)
library(apeglm)
library("pheatmap")
```

#### Upload the data
```{r}
setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Spiders_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")
```


```{r}

#upload count matrix
dat_kallisto<-read.csv("your_reorganized_file_Agosto.csv", 
              sep = ",", header = T, row.names = 1)

kallisto_matrix<-data.matrix(dat_kallisto)
head(dat_kallisto)

```

**Note:** This matrix come from merging the individual matrix from Kallisto with the script `From_Kalisto_To_DEseq_Matrix`. This script Overall, the script takes in expression data for multiple biological samples and outputs a single `TSV file` containing the est counts for each transcript in each sample. The est counts are aggregated across all input files, which makes it easier to compare expression levels between samples and it sorted by columns names.

```{r}
# set work directory
# read in sample info
info3 <-read.csv("sample.csv", sep = "\t", header = T, row.names = 1)
head(info3)
```
```{r}
info3$condition <- factor(info3$condition)

```

**Sample info:** the file `sample.csv` contains additional information about the samples, such as experimental conditions, treatment groups, sample characteristics, or any other relevant metadata associated with each sample. The file is used to specify the experimental design and sample groups for the differential expression analysis in DESeq2.

**It is absolutely critical that the columns of the count matrix `matrix` and the rows of the column data or `sample.csv` (information about samples) are in the same order.**

**DESeq2 will not make guesses as to which column of the count matrix belongs to which row of the column data, these must be provided to DESeq2 already in consistent order.**

```{r}
# Check that the number of columns are the same to the number of row 
all(colnames(kallisto_matrix) %in% rownames(info3))
# are they in the same order?
all(colnames(kallisto_matrix) == rownames(info3))
```

## Construct a DESeqDataSet

```{r}

#construct a DESeqDataSet object

dds_kallisto <- DESeqDataSetFromMatrix(countData = round(kallisto_matrix),
                              colData = info3,
                              design = ~condition)
dds_kallisto
```

### Pre-filtering, yes or no?

While it is not necessary to pre-filter low count genes before running the DESeq2 functions, there are two reasons which make pre-filtering useful.

**1. Memory and Speed Optimization:** By removing rows (genes) with very few reads, pre-filtering reduces the memory size of the DESeq2 data object (referred to as "dds") and improves the speed of transformation and testing functions within DESeq2. This is because working with a smaller dataset requires less computational resources and can lead to faster analysis.

**2. Improved Visualizations:** Pre-filtering can enhance visualizations by excluding features (genes) that have no information for differential expression. Genes with extremely low counts may not provide meaningful insights and can clutter visualizations. Removing them allows for cleaner and more informative visualizations.

Overall, pre-filtering low count genes can improve the efficiency of DESeq2 analysis in terms of memory usage and computational speed. It can also enhance visualizations by excluding genes with negligible expression levels. However, it's essential to consider the appropriate level of filtering based on the specific dataset and analysis objectives.

```{r}
# Remove low gene counts
keep <-rowSums(counts(dds_kallisto)) >= 0
head(keep)
```

```{r}
# then keep the rows in which the previous condition is TRUE,
#remove low expresses genes
dds_kallisto <- dds_kallisto[keep,]
dds_kallisto

```

### Set the factor level

By default, R will choose a reference level for factors based on alphabetical order. Then, if you never tell the DESeq2 functions which level you want to compare against (e.g. which level represents the control group), the comparisons will be based on the alphabetical order of the levels. There are two solutions: you can either explicitly tell results which comparison to make using the contrast argument (this will be shown later), or you can explicitly set the factors levels. In order to see the change of reference levels reflected in the results names, you need to either run DESeq or nbinomWaldTest/nbinomLRT after the re-leveling operation. Setting the factor levels can be done in two ways, either using factor:

```{r}
# To specify the factors levels

dds_kallisto$condition<- relevel(dds_kallisto$condition, ref = "Male_proximal", "Female_proximal") # If the reference level is not mentioned, then it will automatically choose the reference alphabetically
dds_kallisto$condition
```

## Differential expression analysis

The standard differential expression analysis steps are wrapped into a single function, DESeq2. The estimation steps performed by this function are described below, in the manual page for ?DESeq and in the Methods section of the DESeq2 publication (Love, Huber, and Anders 2014).

Results tables are generated using the function results, which extracts a results table with **log2 fold changes**, **p values** and **adjusted p values**. With no additional arguments to results, the log2 fold change and **Wald test p value will be for the last variable in the design formula**, and if this is a factor, the comparison will be the last level of this variable over the reference level (see previous note on factor levels). However, the order of the variables of the design do not matter so long as the user specifies the comparison to build a results table for, using the name or contrast arguments of results.

Details about the comparison are printed to the console, directly above the results table. The text, condition treated vs untreated, tells you that the estimates are of the logarithmic fold change log2(treated/untreated).

```{r}
#run main DESeq program
ddsDE_kallisto<-DESeq(dds_kallisto)
```

## Export normalized read counts

```{r}
normCounts<-counts(ddsDE_kallisto, normalized=T)
```

```{r}
#Export the normalized counts
setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")
write.csv(normCounts, "normalized_data_13_October_Kallisto_Matrix.csv")

```

```{r}
res_kallisto<- results(ddsDE_kallisto)
res_kallisto
```

```{r}
#summary
summary(res_kallisto)
```

```{r}
resultsNames(ddsDE_kallisto)
```

```{r}
# reorder results by padj, 
#44:21 from #https://www.youtube.com/watch?v=wPzeea1Do18&t=812s
setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")
resOrdered_kallisto<-res_kallisto[order(res_kallisto$padj),]
write.csv(resOrdered_kallisto, "resOrdered_padj_Kallisto.csv")
```

Now, let's compare for treatments

### Male_distal vs Male_proximal contrast

Note that it could have specified the coefficient or contrast we want to build a results table for, using either of the following equivalent commands:

```{r}
resKallisto_Treatmenta <- results(ddsDE_kallisto, contrast = c("condition", "Male_distal", "Male_proximal"))
resKallisto_Treatmenta
```

```{r}
summary(resKallisto_Treatmenta)
```

### Volcano Plot Male_distal vs Male_proximal contrast

In DESeq2, the function plotMA shows the log2 fold changes attributable to a given variable over the mean of normalized counts for all the samples in the DESeqDataSet.

Points will be colored red if the adjusted p value is less than 0.1. Points which fall out of the window are plotted as open triangles pointing either up or down.

```{r}
#Volcano plot
plotMA1<-plotMA(resKallisto_Treatmenta, ylim=c(-4,4) , main="Male distal vs Male proximal")
plotMA1
```

### Female_distal vs Female_proximal

```{r}
resKallisto_Treatmentb<- results(ddsDE_kallisto, contrast = c("condition", "Female_distal", "Female_proximal"))
resKallisto_Treatmentb
```

```{r}
# reorder results by padj, 44:21 from #https://www.youtube.com/watch?v=wPzeea1Do18&t=812s

setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")
resOrdered_resKallisto_Treatmentb<-resKallisto_Treatmentb[order(resKallisto_Treatmentb$padj),]
write.csv(resOrdered_resKallisto_Treatmentb, "resOrdered_padj_resKallisto_Treatmentb_10_13.csv")
```

### Volcano plot Female_distal vs Female_proximal

```{r}
#Volcano plot Female_distal vs Female_proximal
plotMA2<-plotMA(resKallisto_Treatmentb, ylim=c(-4,4), main="Female_distal vs Female_proximal")
plotMA2
```

```{r}
summary(resKallisto_Treatmentb)
```

### Male_distal vs Female_distal

```{r}
resKallisto_Treatmentc<- results(ddsDE_kallisto, contrast = c("condition", "Male_distal", "Female_distal"))
resKallisto_Treatmentc
```

```{r}
summary(resKallisto_Treatmentc)
```

```{r}
# reorder results by padj, 44:21 from #https://www.youtube.com/watch?v=wPzeea1Do18&t=812s

setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")

resOrdered_resKallisto_Treatmentb<-resKallisto_Treatmentc[order(resKallisto_Treatmentc$padj),]
write.csv(resOrdered_resKallisto_Treatmentb, "resOrdered_padj_resKallisto_Treatmentc_10_13.csv")
```

### Volcano plot Male_distal vs Female_distal

```{r}
#Volcano plot Male_distal vs Female_distal
plotMA3<-plotMA(resKallisto_Treatmentc, 
                   ylim=c(-4,4), main="Male_distal vs Female_distal")
plotMA3
```

### Male_proximal vs Female_proximal

```{r}

resKallisto_Treatmentd<- results(ddsDE_kallisto, 
                           contrast = c("condition", "Male_proximal", "Female_proximal"))
resKallisto_Treatmentd
```

```{r}
summary(resKallisto_Treatmentd)
```

```{r}
# reorder results by padj, 44:21 from #https://www.youtube.com/watch?v=wPzeea1Do18&t=812s

setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")

resOrdered_resKallisto_Treatmentd<-resKallisto_Treatmentd[order(resKallisto_Treatmentd$padj),]
write.csv(resOrdered_resKallisto_Treatmentd, "resOrdered_padj_resKallisto_Treatmentd_10_13.csv")

```

### Volcano plot Male_proximal vs Female_proximal

```{r}
#Volcano plot Male_distal vs Female_distal
plotMA4<-plotMA(resKallisto_Treatmentd, 
                   ylim=c(-4,4), main="Male_proximal vs Female_proximal")
plotMA4

```

### Daniella

Hacer un contraste entre el que me falta comparar male proximal vs female distal

### All PlotMA

```{r}
# Set up a multi-panel plot
par(mfrow = c(2, 2))

# Plot 1
plotMA(resKallisto_Treatmentd, ylim=c(-4, 4), main="Male_proximal vs Female_proximal")

# Plot 2
plotMA(resKallisto_Treatmentc, ylim=c(-4, 4), main="Male_distal vs Female_distal")

# Plot 3
plotMA(resKallisto_Treatmentb, ylim=c(-4, 4), main="Female_distal vs Female_proximal")

# Plot 4
plotMA(resKallisto_Treatmenta, ylim=c(-4, 4), main="Male_distal vs Male_proximal")

# Reset the plotting settings
par(mfrow = c(1, 1))

```

### Plot counts, To explore the specific receptors

To examine the counts of reads for a single gene across the groups. A simple function for making this plot is plotCounts, which normalizes counts by the estimated size factors (or normalization factors if these were used) and adds a pseudocount of 1/2 to allow for log scale plotting. The counts are grouped by the variables in intgroup, where more than one variable can be specified. Here we specify the gene which had the smallest p value from the results table created above.

You can select the gene to plot by rowname or by numeric index.

```{r}

plotCounts(ddsDE_kallisto, gene=which.min(res_kallisto$padj), intgroup="condition")

```

For customized plotting, an argument `returnData` specifies that the function should only return a data.frame for plotting with ggplot.

The function `plotCounts` in DESeq2 is used to visualize the count data for a particular gene or set of genes. It allows you to explore the raw count values for individual genes across different conditions or samples in your dataset. Here's an explanation of the function and its parameters:

```{r}
d_kallisto <- plotCounts(ddsDE_kallisto, gene=which.min(res_kallisto$padj), intgroup="condition", returnData=TRUE)

library("ggplot2")
ggplot(d_kallisto, aes(x=condition, y=count)) + 
  geom_point(position=position_jitter(w=0.1,h=0)) + 
  scale_y_log10(breaks=c(25,100,400))
```

### Visualization

```{r}
# normCounts

setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Spiders_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")
normCounts_Kallisto <-read.csv("normalized_data_13_October_Kallisto_Matrix.csv", 
                         sep = ",", header = T, row.names = 1)

```

```{r}
setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Spiders_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")

deSeqRes_A_Kallisto<-read.csv("resOrdered_padj_resKallisto_Treatmenta_10_13.csv", 
                        sep = ",", header = T, row.names = 1)

```

```{r}
#remove NA from res_Treatmenta
deSeqRes_A_Kallisto<- na.omit(resKallisto_Treatmenta)
```

```{r}
# Adding a new column of significance 

deSeqRes_A_Kallisto$sig <- ifelse(deSeqRes_A_Kallisto$padj <= 0.1, "yes", "no" )
head(deSeqRes_A_Kallisto)
deSeqRes_A_Kallisto<-data.frame(deSeqRes_A_Kallisto)
view(deSeqRes_A_Kallisto)

```

#### Create csv file for each treatment

##### Treatment a

```{r}

setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Spiders_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")
# Heat map, I plot only the genes that have a p value mas than 0.1
# subset for res_Treatmenta 

signi_A_Kallisto<-subset(resKallisto_Treatmenta, padj <= 0.1)
#allsi<-merge(normCounts, signi, by = 0)
signi_A_Kallisto<-data.frame(signi_A_Kallisto)
```

```{r}

allsig_A_Kallisto<-merge(normCounts_Kallisto, signi_A_Kallisto, by=0)

```

```{r}
#row and columns
#sigCounts <- allsi[1:20,1:96]
sigCounts_A_Kallisto <- allsig_A_Kallisto[,2:97]
```

```{r}
row.names(sigCounts_A_Kallisto) <-allsig_A_Kallisto$Row.names
```

```{r}
 
# Define the desired order of the columns
order_column <- c( 
"S412_S1_L001" , "S412_S1_L002" , "S412_S1_L003" , "S412_S1_L004" , 
"S413_S2_L001" , "S413_S2_L002" , "S413_S2_L003" , "S413_S2_L004" , 
"S414_S3_L001" , "S414_S3_L002" , "S414_S3_L003" , "S414_S3_L004" , 
"S415_S4_L001" , "S415_S4_L002" , "S415_S4_L003" , "S415_S4_L004" , 
"S416_S5_L001" , "S416_S5_L002" , "S416_S5_L003" , "S416_S5_L004" , 
"S417_S6_L001" , "S417_S6_L002" , "S417_S6_L003" , "S417_S6_L004" , 
"S418_S7_L001" , "S418_S7_L002" , "S418_S7_L003" , "S418_S7_L004" , 
"S419_S8_L001" , "S419_S8_L002" , "S419_S8_L003" , "S419_S8_L004" , 
"S420_S9_L001" , "S420_S9_L002" , "S420_S9_L003" , "S420_S9_L004" , 
"S421_S10_L001", "S421_S10_L002", "S421_S10_L003", "S421_S10_L004", 
"S422_S11_L001", "S422_S11_L002", "S422_S11_L003", "S422_S11_L004", 
"S423_S12_L001", "S423_S12_L002", "S423_S12_L003", "S423_S12_L004", 
"S496_S1_L001" , "S496_S1_L002" , "S496_S1_L003" , "S496_S1_L004" , 
"S497_S2_L001" , "S497_S2_L002" , "S497_S2_L003" , "S497_S2_L004" , 
"S498_S3_L001" , "S498_S3_L002" , "S498_S3_L003" , "S498_S3_L004" , 
"S499_S4_L001" , "S499_S4_L002" , "S499_S4_L003" , "S499_S4_L004" , 
"S500_S5_L001" , "S500_S5_L002" , "S500_S5_L003" , "S500_S5_L004" , 
"S501_S6_L001" , "S501_S6_L002" , "S501_S6_L003" , "S501_S6_L004" , 
"S502_S7_L001" , "S502_S7_L002" , "S502_S7_L003" , "S502_S7_L004" , 
"S503_S8_L001" , "S503_S8_L002" , "S503_S8_L003" , "S503_S8_L004" , 
"S504_S9_L001" , "S504_S9_L002" , "S504_S9_L003" , "S504_S9_L004" , 
"S505_S10_L001", "S505_S10_L002", "S505_S10_L003", "S505_S10_L004", 
"S506_S11_L001", "S506_S11_L002", "S506_S11_L003", "S506_S11_L004", 
"S507_S12_L001", "S507_S12_L002", "S507_S12_L003", "S507_S12_L004")

```

```{r}
setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Spiders_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")

# Reordena la matriz de datos según el orden deseado de las columnas
matrix_dat_order_A_Kallisto <- sigCounts_A_Kallisto[, order_column]

```

```{r}
#Export the new data
setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Spiders_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")

write.csv(matrix_dat_order_A_Kallisto, "matrix_dat_order_A_Kallisto.csv")

```

##### Tratment b

```{r}
setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Spiders_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")

# subset for res_Treatmenta 

signi_B_Kallisto<-subset(resKallisto_Treatmentb, padj <= 0.1)
#allsi<-merge(normCounts, signi, by = 0)
signi_B_Kallisto<-data.frame(signi_B_Kallisto)
```

```{r}
allsig_B_Kallisto<-merge(normCounts_Kallisto, signi_B_Kallisto, by=0)
```

```{r}
#row and columns
#sigCounts <- allsi[1:20,1:96]
sigCounts_B_Kallisto <- allsig_B_Kallisto[,2:97]
```

```{r}
row.names(sigCounts_B_Kallisto) <-allsig_B_Kallisto$Row.names
```

```{r}
# Reordena la matriz de datos según el orden deseado de las columnas
matrix_dat_order_B_Kallisto <- sigCounts_B_Kallisto[, order_column]
```

```{r}
#Export the new data
setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Spiders_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")

write.csv(matrix_dat_order_B_Kallisto, "matrix_dat_order_B_Kallisto.csv")

```

##### Tratment c

```{r}
# subset for res_Treatmenta 
setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Spiders_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")

signi_C_Kallisto<-subset(resKallisto_Treatmentc, padj <= 0.1)
#allsi<-merge(normCounts, signi, by = 0)
signi_C_Kallisto<-data.frame(signi_C_Kallisto)
```

```{r}
allsig_C_Kallisto<-merge(normCounts_Kallisto, signi_C_Kallisto, by=0)
```

```{r}
#row and columns
#sigCounts <- allsi[1:20,1:96]
sigCounts_C_Kallisto <- allsig_C_Kallisto[,2:97]
```

```{r}
row.names(sigCounts_C_Kallisto) <-allsig_C_Kallisto$Row.names
```

```{r}
# Reordena la matriz de datos según el orden deseado de las columnas
matrix_dat_order_C_Kallisto <- sigCounts_C_Kallisto[, order_column]
```

```{r}
setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Spiders_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")

write.csv(matrix_dat_order_C_Kallisto, "matrix_dat_order_C_Kallisto.csv")
```

##### Tratment d

```{r}
# subset for res_Treatmenta 
setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Spiders_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")

signi_D_Kallisto<-subset(resKallisto_Treatmentd, padj <= 0.1)
#allsi<-merge(normCounts, signi, by = 0)
signi_D_Kallisto<-data.frame(signi_D_Kallisto)
```

```{r}
allsig_D_Kallisto<-merge(normCounts_Kallisto, signi_D_Kallisto, by=0)
```

```{r}
#row and columns
#sigCounts <- allsi[1:20,1:96]
sigCounts_D_Kallisto <- allsig_D_Kallisto[,2:97]
```

```{r}
row.names(sigCounts_D_Kallisto) <-allsig_D_Kallisto$Row.names
```

```{r}
# Reordena la matriz de datos según el orden deseado de las columnas
matrix_dat_order_D_Kallisto <- sigCounts_D_Kallisto[, order_column]

```

```{r}
setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Spiders_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")

#Export the new data
write.csv(matrix_dat_order_D_Kallisto, "matrix_dat_order_D_Kallisto.csv")
```

I merged all the `matrix_dat_order_N_GR.csv` and then I removed the duplicated rows to create the `matrix_dat_order_All_No_Duplicates_GR.csv` file.

`matrix_dat_order_All_Duplicates_Kallisto.csv`

\`cat matrix_dat_order_A_GR.csv matrix_dat_order_B_GR.csv matrix_dat_order_C_GR.csv matrix_dat_order_D_GR.csv \> matrix_dat_order_All_GR.csv\`\`

**To remove duplicated**

1.  Print Duplicated Rows: sort input.csv \| uniq -d \> duplicates.csv

2.  Remove Duplicated Rows: sort input.csv \| uniq \> output.csv

### Exporting files with up or downregulated information

```{r}
setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Spiders_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")

# Extract the gene names - up or downregulated 

gene_names <- rownames(resKallisto_Treatmentd)

# Extract the summary information
log2FoldChange <- resKallisto_Treatmentd$log2FoldChange
padj <- resKallisto_Treatmentd$padj
baseMean <- resKallisto_Treatmentd$baseMean
lfcSE <- resKallisto_Treatmentd$lfcSE
stat <- resKallisto_Treatmentd$stat

# Create a data frame with the summary information and gene names
summary_table_Kallisto_D <- data.frame(Gene = gene_names,
                            log2FoldChange = log2FoldChange,
                            padj = padj,
                            baseMean = baseMean,
                            lfcSE = lfcSE,
                            stat = stat)

# Add a column indicating upregulation or downregulation
summary_table_Kallisto_D$Regulation <- ifelse(summary_table_Kallisto_D$log2FoldChange > 0,
                                         "Upregulated", "Downregulated")

# Export the summary table as a CSV file
write.csv(summary_table_Kallisto_C, file = "summary_results_Kallisto_D.csv", 
          row.names = FALSE)
```

```{r}
#upload the merge data
setwd("/Users/zaidemontesortiz/Documents/1_PhD_project_Spiders_Ticks_Beetles/Spiders/Results/Abundance_Matrix/Abundance_Original")
#upload count matrix
dat_merge_all_Kallisto<-read.csv("matrix_dat_order_All_No_Duplicates_Kallisto.csv", 
            sep = ",", header = T, row.names = 1)
```

```{r}
head(dat_merge_all_Kallisto)
```

### HeatMap iGluRs and IRs

```{r}
# Filter the rows in dat_merge_all_Kallisto based on significant genes belonging to a specific family
specific_gene_family<- c("STRG.36829.3.p2", "STRG.63751.1.p1", "STRG.32492.1.p2", "STRG.29635.1.p1", "STRG.24254.2.p1", "STRG.24262.3.p1", "STRG.38773.1.p1", "STRG.49562.1.p1", "STRG.33523.1.p1", "STRG.33525.2.p1", "STRG.33525.1.p1", "STRG.50993.1.p1", "STRG.26518.1.p1", "STRG.50071.1.p1", "STRG.26560.2.p3", "STRG.14809.2.p1", "STRG.26520.1.p4", "STRG.26541.3.p2", "STRG.33350.1.p1", "STRG.45426.1.p1", "STRG.12677.1.p2", "STRG.62762.1.p1", "STRG.21210.1.p1", "STRG.2167.1.p1", "STRG.63201.1.p1", "STRG.43176.1.p1", "STRG.15456.1.p1", "STRG.37954.1.p1", "STRG.39111.1.p2", "STRG.33530.1.p1", "STRG.33519.1.p3", "STRG.26560.1.p3", "STRG.46044.1.p1", "STRG.49563.1.p1", "STRG.56704.4.p1", "STRG.56704.1.p1", "STRG.52863.1.p1", "STRG.46275.1.p2", "STRG.46044.3.p1", "STRG.46044.2.p1", "STRG.50985.1.p1", "STRG.46031.3.p1", "STRG.56704.3.p1", "STRG.56704.2.p1", "STRG.7561.1.p2", "STRG.56704.5.p1", "STRG.2001.1.p1", "STRG.19212.3.p1", "STRG.18882.2.p1", "STRG.39429.2.p1", "STRG.33350.2.p1", "STRG.19212.2.p1", "STRG.26540.1.p1", "STRG.65166.1.p1", "STRG.33530.2.p2", "STRG.38773.1.p2", "STRG.33519.2.p1", "STRG.29634.1.p1", "STRG.11765.1.p1", "STRG.12677.1.p1", "STRG.20575.1.p7", "STRG.63712.1.p1", "STRG.38406.1.p1", "STRG.35168.1.p1", "STRG.26541.2.p1", "STRG.13473.1.p1", "STRG.5450.2.p1", "STRG.1974.1.p1", "STRG.50992.1.p1", "STRG.35168.2.p1", "STRG.29634.2.p1", "STRG.11600.1.p2", "STRG.11765.2.p1", "STRG.11767.1.p1", "STRG.11770.2.p1", "STRG.12825.1.p2", "STRG.13469.1.p1", "STRG.1588.1.p1", "STRG.18882.1.p1", "STRG.20574.1.p2", "STRG.21136.2.p1", "STRG.2159.1.p1", "STRG.27354.1.p1", "STRG.29632.1.p1", "STRG.32906.1.p1", "STRG.32908.1.p3", "STRG.33520.1.p1", "STRG.38401.1.p1", "STRG.38402.1.p1", "STRG.38405.1.p1", "STRG.4504.2.p1", "STRG.45427.1.p1", "STRG.45427.2.p1", "STRG.50989.1.p2", "STRG.56820.4.p1", "STRG.61662.1.p1", "STRG.63752.1.p1", "STRG.63755.1.p1", "STRG.7818.1.p1", "STRG.26555.1.p1", "STRG.46031.1.p1", "STRG.52863.2.p1", "STRG.52918.5.p1", "STRG.52927.1.p1", "STRG.52929.1.p1", "STRG.5450.2.p2", "STRG.5450.2.p3", "STRG.56703.1.p1", "STRG.65166.2.p1", "STRG.7545.1.p1", "STRG.7819.1.p1"
)  # Replace with your specific gene family

```

```{r}
significant_genes <- rownames(dat_merge_all_Kallisto) %in% specific_gene_family

```

```{r}

# Define the condition table with the column names and the corresponding conditions
tabla_condiciones <- data.frame(
  sample = colnames(dat_merge_all_Kallisto),
  condition = c(
    "Male_distal", "Male_distal", "Male_distal", "Male_distal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Male_distal", "Male_distal",
    "Male_distal", "Male_distal", "Male_proximal", "Male_proximal", "Male_proximal",
    "Male_proximal", "Male_distal", "Male_distal", "Male_distal", "Male_distal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Male_proximal", "Male_distal",
    "Male_distal", "Male_distal", "Male_distal", "Male_proximal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_distal", "Male_distal", "Male_distal",
    "Male_distal", "Male_proximal", "Male_proximal", "Male_proximal", "Male_proximal",
    "Male_distal", "Male_distal", "Male_distal", "Male_distal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Female_distal", "Female_distal",
    "Female_distal", "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_proximal", "Female_distal", "Female_distal", "Female_distal", "Female_distal",
    "Female_proximal", "Female_proximal", "Female_proximal", "Female_proximal", "Female_distal",
    "Female_distal", "Female_distal", "Female_distal", "Female_proximal", "Female_proximal",
    "Female_proximal", "Female_proximal", "Female_distal", "Female_distal", "Female_distal",
    "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_distal", "Female_distal", "Female_distal", "Female_distal", "Female_proximal",
    "Female_proximal", "Female_proximal", "Female_proximal", "Female_distal", "Female_distal",
    "Female_distal", "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_proximal"
  )
)


```

```{r}
colnames(dat_merge_all_Kallisto) <- tabla_condiciones$condition
```

```{r}
#col_order <- order(duplicated(colnames(matrix_dat_order)), colnames(matrix_dat_order))


col_order <- order(colnames(dat_merge_all_Kallisto), colnames(dat_merge_all_Kallisto), duplicated(colnames(dat_merge_all_Kallisto)))
```

```{r}
# Subsetting the data for significant genes in the specified gene family

subset_data <- dat_merge_all_Kallisto[significant_genes, col_order]

```

```{r}
# Plotting heatmap for the specific gene family
pheatmap(log2(subset_data + 1), cluster_cols = FALSE, cluster_rows = FALSE, scale = 'none', gaps_col = c(24, 48, 72, 96))
```

### Heatmap GR/GPCRs

```{r}
# Filter the rows in dat_merge_all_Kallisto based on significant genes belonging to a specific family
specific_gene_family_GR<- c("STRG.10027.1.p1", "STRG.10099.2.p1", "STRG.10100.2.p1", "STRG.11035.1.p1", "STRG.11903.1.p2", "STRG.11903.2.p1", "STRG.12628.1.p1", "STRG.13962.1.p1", "STRG.14877.1.p2", "STRG.174.2.p1", "STRG.19037.1.p2", "STRG.19057.1.p1", "STRG.19222.1.p1", "STRG.19234.2.p2", "STRG.19234.4.p1", "STRG.19785.1.p1", "STRG.19939.1.p3", "STRG.20413.1.p2", "STRG.20477.1.p1", "STRG.21709.1.p1", "STRG.21772.4.p1", "STRG.218.1.p1", "STRG.21836.1.p1", "STRG.21837.1.p1", "STRG.22259.1.p2", "STRG.22259.3.p4", "STRG.22577.1.p3", "STRG.22782.1.p1", "STRG.24009.1.p1", "STRG.24530.1.p2", "STRG.24531.1.p2", "STRG.24533.1.p1", "STRG.24800.2.p1", "STRG.25665.1.p1", "STRG.25665.1.p2", "STRG.26574.1.p2", "STRG.27445.1.p2", "STRG.29402.1.p1", "STRG.29402.3.p1", "STRG.30251.1.p1", "STRG.30251.2.p1", "STRG.31577.1.p1", "STRG.32503.2.p1", "STRG.32549.1.p1", "STRG.32697.1.p2", "STRG.32700.1.p2", "STRG.32745.1.p1", "STRG.32805.1.p2", "STRG.32819.1.p1", "STRG.33058.3.p1", "STRG.33106.1.p1", "STRG.33113.1.p1", "STRG.33122.2.p1", "STRG.34226.1.p2", "STRG.34226.3.p1", "STRG.34228.1.p1", "STRG.34228.1.p2", "STRG.35198.1.p2", "STRG.35200.1.p1", "STRG.35277.1.p1", "STRG.35279.1.p2", "STRG.35285.1.p1", "STRG.35905.1.p1", "STRG.36502.1.p1", "STRG.36505.1.p1", "STRG.36825.1.p1", "STRG.37440.1.p1", "STRG.37607.1.p1", "STRG.37608.1.p1", "STRG.37800.1.p2", "STRG.38913.1.p2", "STRG.44446.1.p1", "STRG.44692.1.p1", "STRG.44966.1.p1", "STRG.45115.1.p2", "STRG.45197.1.p1", "STRG.45197.1.p2", "STRG.45228.1.p1", "STRG.45601.1.p1", "STRG.45731.1.p1", "STRG.48414.1.p2", "STRG.50419.1.p1", "STRG.50770.1.p1", "STRG.50770.2.p1", "STRG.52153.1.p4", "STRG.52153.2.p1", "STRG.52154.1.p1", "STRG.54336.1.p1", "STRG.55966.1.p1", "STRG.55966.1.p2", "STRG.56029.1.p1", "STRG.56030.1.p1", "STRG.56530.1.p1", "STRG.59085.1.p1", "STRG.60723.1.p1", "STRG.62648.1.p1", "STRG.62956.1.p1", "STRG.63473.1.p3", "STRG.63515.1.p1", "STRG.63816.1.p1", "STRG.63863.1.p1", "STRG.65723.1.p1", "STRG.67512.1.p1", "STRG.6793.1.p2", "STRG.68983.1.p1", "STRG.69382.1.p1", "STRG.69962.1.p1", "STRG.72024.1.p2", "STRG.7817.1.p1", "STRG.8704.1.p1", "STRG.8964.1.p1"

)  # Replace with your specific gene family

```

```{r}
significant_genes_GR <- rownames(dat_merge_all_Kallisto) %in% specific_gene_family_GR

```

```{r}


library(pheatmap)

# Define la tabla de condiciones con los nombres de columna y las condiciones correspondientes
tabla_condiciones <- data.frame(
  sample = colnames(dat_merge_all_Kallisto),
  condition = c(
    "Male_distal", "Male_distal", "Male_distal", "Male_distal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Male_distal", "Male_distal",
    "Male_distal", "Male_distal", "Male_proximal", "Male_proximal", "Male_proximal",
    "Male_proximal", "Male_distal", "Male_distal", "Male_distal", "Male_distal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Male_proximal", "Male_distal",
    "Male_distal", "Male_distal", "Male_distal", "Male_proximal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_distal", "Male_distal", "Male_distal",
    "Male_distal", "Male_proximal", "Male_proximal", "Male_proximal", "Male_proximal",
    "Male_distal", "Male_distal", "Male_distal", "Male_distal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Female_distal", "Female_distal",
    "Female_distal", "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_proximal", "Female_distal", "Female_distal", "Female_distal", "Female_distal",
    "Female_proximal", "Female_proximal", "Female_proximal", "Female_proximal", "Female_distal",
    "Female_distal", "Female_distal", "Female_distal", "Female_proximal", "Female_proximal",
    "Female_proximal", "Female_proximal", "Female_distal", "Female_distal", "Female_distal",
    "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_distal", "Female_distal", "Female_distal", "Female_distal", "Female_proximal",
    "Female_proximal", "Female_proximal", "Female_proximal", "Female_distal", "Female_distal",
    "Female_distal", "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_proximal"
  )
)


```

```{r}
colnames(dat_merge_all_Kallisto) <- tabla_condiciones$condition
```

```{r}
#col_order <- order(duplicated(colnames(matrix_dat_order)), colnames(matrix_dat_order))


col_order <- order(colnames(dat_merge_all_Kallisto), colnames(dat_merge_all_Kallisto), duplicated(colnames(dat_merge_all_Kallisto)))
```

```{r}
# Subsetting the data for significant genes in the specified gene family

subset_data_GR <- dat_merge_all_Kallisto[significant_genes_GR, col_order]

```

```{r}
# Plotting heatmap for the specific gene family
pheatmap(log2(subset_data_GR + 1), cluster_cols = FALSE, cluster_rows = FALSE, scale = 'none', gaps_col = c(24, 48, 72, 96))
```

### Heatmap Chemotactile receprots

```{r}
# Filter the rows in dat_merge_all_Kallisto based on significant genes belonging to a specific family
specific_gene_family_CTR<- c("STRG.310.1.p2", "STRG.31837.2.p1", "STRG.33283.3.p2", "STRG.36356.6.p1", "STRG.36816.2.p1", "STRG.36816.3.p1", "STRG.36816.6.p1", "STRG.39168.2.p1", "STRG.44106.1.p1", "STRG.44679.2.p1", "STRG.4521.1.p1", "STRG.45749.1.p2", "STRG.46019.1.p2", "STRG.46019.3.p1", "STRG.47451.2.p1", "STRG.48506.1.p1", "STRG.50053.1.p4", "STRG.50645.1.p1", "STRG.50892.1.p2", "STRG.52542.1.p2", "STRG.52542.2.p1", "STRG.52914.1.p1", "STRG.5675.1.p1", "STRG.5686.2.p1", "STRG.5689.1.p2", "STRG.59144.1.p2", "STRG.59628.2.p1", "STRG.61673.1.p1", "STRG.61675.1.p1", "STRG.7109.1.p2", "STRG.7111.1.p4", "STRG.7263.4.p1"


)  # Replace with your specific gene family

```

```{r}
significant_genes_CTR <- rownames(dat_merge_all_Kallisto) %in% specific_gene_family_CTR

```

```{r}

library(pheatmap)

# Define la tabla de condiciones con los nombres de columna y las condiciones correspondientes
tabla_condiciones <- data.frame(
  sample = colnames(dat_merge_all_Kallisto),
  condition = c(
    "Male_distal", "Male_distal", "Male_distal", "Male_distal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Male_distal", "Male_distal",
    "Male_distal", "Male_distal", "Male_proximal", "Male_proximal", "Male_proximal",
    "Male_proximal", "Male_distal", "Male_distal", "Male_distal", "Male_distal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Male_proximal", "Male_distal",
    "Male_distal", "Male_distal", "Male_distal", "Male_proximal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_distal", "Male_distal", "Male_distal",
    "Male_distal", "Male_proximal", "Male_proximal", "Male_proximal", "Male_proximal",
    "Male_distal", "Male_distal", "Male_distal", "Male_distal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Female_distal", "Female_distal",
    "Female_distal", "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_proximal", "Female_distal", "Female_distal", "Female_distal", "Female_distal",
    "Female_proximal", "Female_proximal", "Female_proximal", "Female_proximal", "Female_distal",
    "Female_distal", "Female_distal", "Female_distal", "Female_proximal", "Female_proximal",
    "Female_proximal", "Female_proximal", "Female_distal", "Female_distal", "Female_distal",
    "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_distal", "Female_distal", "Female_distal", "Female_distal", "Female_proximal",
    "Female_proximal", "Female_proximal", "Female_proximal", "Female_distal", "Female_distal",
    "Female_distal", "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_proximal"
  )
)


```

```{r}
colnames(dat_merge_all_Kallisto) <- tabla_condiciones$condition
```

```{r}
#col_order <- order(duplicated(colnames(matrix_dat_order)), colnames(matrix_dat_order))


col_order <- order(colnames(dat_merge_all_Kallisto), colnames(dat_merge_all_Kallisto), duplicated(colnames(dat_merge_all_Kallisto)))
```

```{r}
# Subsetting the data for significant genes in the specified gene family

subset_data_CTR <- dat_merge_all_Kallisto[significant_genes_CTR, col_order]

```

```{r}
# Plotting heatmap for the specific gene family
pheatmap(log2(subset_data_CTR + 1), cluster_cols = FALSE, cluster_rows = FALSE, scale = 'none', gaps_col = c(24, 48, 72, 96))
```

### Heatmap G-Proteins

```{r}
# Filter the rows in dat_merge_all_Kallisto based on significant genes belonging to a specific family
specific_gene_family_Gprot<- c("STRG.10628.1.p1","STRG.15081.2.p1","STRG.15081.3.p1","STRG.22550.1.p1","STRG.27469.1.p1","STRG.30974.2.p1","STRG.35888.1.p2","STRG.40611.1.p1","STRG.41251.1.p2","STRG.41251.2.p1","STRG.42595.1.p1","STRG.42595.2.p1","STRG.42873.1.p2","STRG.45112.2.p1","STRG.46364.1.p1","STRG.55745.1.p1","STRG.60555.1.p1","STRG.70937.1.p1","STRG.728.1.p2") 
# Replace with your specific gene family

```

```{r}
significant_genes_Gprot <- rownames(dat_merge_all_Kallisto) %in% specific_gene_family_Gprot
```

```{r}

library(pheatmap)

# Define la tabla de condiciones con los nombres de columna y las condiciones correspondientes
tabla_condiciones <- data.frame(
  sample = colnames(dat_merge_all_Kallisto),
  condition = c(
    "Male_distal", "Male_distal", "Male_distal", "Male_distal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Male_distal", "Male_distal",
    "Male_distal", "Male_distal", "Male_proximal", "Male_proximal", "Male_proximal",
    "Male_proximal", "Male_distal", "Male_distal", "Male_distal", "Male_distal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Male_proximal", "Male_distal",
    "Male_distal", "Male_distal", "Male_distal", "Male_proximal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_distal", "Male_distal", "Male_distal",
    "Male_distal", "Male_proximal", "Male_proximal", "Male_proximal", "Male_proximal",
    "Male_distal", "Male_distal", "Male_distal", "Male_distal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Female_distal", "Female_distal",
    "Female_distal", "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_proximal", "Female_distal", "Female_distal", "Female_distal", "Female_distal",
    "Female_proximal", "Female_proximal", "Female_proximal", "Female_proximal", "Female_distal",
    "Female_distal", "Female_distal", "Female_distal", "Female_proximal", "Female_proximal",
    "Female_proximal", "Female_proximal", "Female_distal", "Female_distal", "Female_distal",
    "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_distal", "Female_distal", "Female_distal", "Female_distal", "Female_proximal",
    "Female_proximal", "Female_proximal", "Female_proximal", "Female_distal", "Female_distal",
    "Female_distal", "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_proximal"
  )
)


```

```{r}
colnames(dat_merge_all_Kallisto) <- tabla_condiciones$condition
```

```{r}
#col_order <- order(duplicated(colnames(matrix_dat_order)), colnames(matrix_dat_order))


col_order <- order(colnames(dat_merge_all_Kallisto), colnames(dat_merge_all_Kallisto), duplicated(colnames(dat_merge_all_Kallisto)))
```

```{r}
# Subsetting the data for significant genes in the specified gene family

subset_data_Gprotein <- dat_merge_all_Kallisto[significant_genes_Gprot, col_order]

```

```{r}
# Plotting heatmap for the specific gene family
pheatmap(log2(subset_data_Gprotein + 1), cluster_cols = FALSE, cluster_rows = FALSE, scale = 'none', gaps_col = c(24, 48, 72, 96))
```

### Heatmap CFTR

```{r}
# Filter the rows in dat_merge_all_Kallisto based on significant genes belonging to a specific family
specific_gene_family_CFTR<- c("STRG.110.1.p2","STRG.21529.1.p1","STRG.2468.1.p1","STRG.56052.1.p1","STRG.56801.1.p2","STRG.63945.1.p1","STRG.66140.1.p1") 
# Replace with your specific gene family

```

```{r}
significant_genes_CFTR <- rownames(dat_merge_all_Kallisto) %in% specific_gene_family_CFTR
```

```{r}

library(pheatmap)

# Define la tabla de condiciones con los nombres de columna y las condiciones correspondientes
tabla_condiciones <- data.frame(
  sample = colnames(dat_merge_all_Kallisto),
  condition = c(
    "Male_distal", "Male_distal", "Male_distal", "Male_distal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Male_distal", "Male_distal",
    "Male_distal", "Male_distal", "Male_proximal", "Male_proximal", "Male_proximal",
    "Male_proximal", "Male_distal", "Male_distal", "Male_distal", "Male_distal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Male_proximal", "Male_distal",
    "Male_distal", "Male_distal", "Male_distal", "Male_proximal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_distal", "Male_distal", "Male_distal",
    "Male_distal", "Male_proximal", "Male_proximal", "Male_proximal", "Male_proximal",
    "Male_distal", "Male_distal", "Male_distal", "Male_distal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Female_distal", "Female_distal",
    "Female_distal", "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_proximal", "Female_distal", "Female_distal", "Female_distal", "Female_distal",
    "Female_proximal", "Female_proximal", "Female_proximal", "Female_proximal", "Female_distal",
    "Female_distal", "Female_distal", "Female_distal", "Female_proximal", "Female_proximal",
    "Female_proximal", "Female_proximal", "Female_distal", "Female_distal", "Female_distal",
    "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_distal", "Female_distal", "Female_distal", "Female_distal", "Female_proximal",
    "Female_proximal", "Female_proximal", "Female_proximal", "Female_distal", "Female_distal",
    "Female_distal", "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_proximal"
  )
)


```

```{r}
colnames(dat_merge_all_Kallisto) <- tabla_condiciones$condition
```

```{r}
#col_order <- order(duplicated(colnames(matrix_dat_order)), colnames(matrix_dat_order))


col_order <- order(colnames(dat_merge_all_Kallisto), colnames(dat_merge_all_Kallisto), duplicated(colnames(dat_merge_all_Kallisto)))
```

```{r}
# Subsetting the data for significant genes in the specified gene family

subset_data_CFTR <- dat_merge_all_Kallisto[significant_genes_CFTR, col_order]

```

```{r}
# Plotting heatmap for the specific gene family
pheatmap(log2(subset_data_CFTR + 1), cluster_cols = FALSE, cluster_rows = FALSE, scale = 'none', gaps_col = c(24, 48, 72, 96))
```

### Heatmap GPCRs

```{r}
# Filter the rows in dat_merge_all_Kallisto based on significant genes belonging to a specific family
specific_gene_family_GPCRs<- c("STRG.10056.2.p1","STRG.10099.2.p1","STRG.10100.2.p1","STRG.11033.2.p1","STRG.11903.1.p2","STRG.11903.2.p1","STRG.12298.1.p2","STRG.12628.1.p1","STRG.12632.2.p1","STRG.13919.1.p1","STRG.13962.1.p1","STRG.15334.3.p1","STRG.15339.2.p1","STRG.15641.1.p2","STRG.15703.2.p1","STRG.15720.1.p1","STRG.15720.1.p2","STRG.15742.1.p2","STRG.15756.1.p1","STRG.16407.2.p1","STRG.174.2.p1","STRG.17621.2.p1","STRG.17621.3.p1","STRG.17758.1.p3","STRG.19037.1.p2","STRG.19057.1.p1","STRG.1908.1.p1","STRG.1919.2.p1","STRG.1919.5.p1","STRG.1919.7.p1","STRG.19234.2.p2","STRG.19234.4.p1","STRG.19386.1.p1","STRG.1959.2.p1","STRG.20075.2.p2","STRG.21180.1.p1","STRG.21181.1.p1","STRG.21306.1.p1","STRG.21729.2.p1","STRG.21729.8.p1","STRG.21745.1.p1","STRG.218.1.p1","STRG.22259.1.p2","STRG.22259.3.p4","STRG.22577.1.p3","STRG.22790.1.p2","STRG.22984.1.p2","STRG.22984.2.p3","STRG.23026.1.p1","STRG.2549.1.p2","STRG.25868.1.p1","STRG.26574.1.p2","STRG.2683.2.p1","STRG.2683.4.p1","STRG.27804.1.p1","STRG.28054.1.p1","STRG.28301.1.p1","STRG.29220.2.p1","STRG.29220.3.p1","STRG.29220.4.p1","STRG.29402.1.p1","STRG.29402.3.p1","STRG.30251.1.p1","STRG.30251.2.p1","STRG.31053.1.p1","STRG.33058.3.p1","STRG.34463.3.p1","STRG.37952.2.p2","STRG.40152.2.p1","STRG.40152.3.p1","STRG.40515.1.p3","STRG.40515.3.p1","STRG.40527.1.p1","STRG.40530.1.p1","STRG.4144.1.p1","STRG.4190.1.p1","STRG.42507.1.p1","STRG.44758.1.p1","STRG.44785.2.p1","STRG.44793.2.p1","STRG.45731.1.p1","STRG.47392.1.p1","STRG.47392.2.p1","STRG.47392.5.p1","STRG.47585.1.p2","STRG.50419.1.p1","STRG.50770.1.p1","STRG.50770.2.p1","STRG.51221.1.p3","STRG.52153.1.p4","STRG.52153.2.p1","STRG.52154.1.p1","STRG.52506.1.p2","STRG.55328.2.p1","STRG.55378.1.p1","STRG.55603.1.p1","STRG.56152.1.p1","STRG.57240.1.p1","STRG.60046.1.p1","STRG.60053.1.p1","STRG.60054.1.p1","STRG.60064.1.p1","STRG.60074.1.p1","STRG.60099.1.p1","STRG.60635.1.p1","STRG.65723.1.p1","STRG.66160.1.p1","STRG.68535.1.p1","STRG.7634.2.p1") 
# Replace with your specific gene family

```

```{r}
significant_genes_GPCRs <- rownames(dat_merge_all_Kallisto) %in% specific_gene_family_GPCRs
```

```{r}

library(pheatmap)

# Define la tabla de condiciones con los nombres de columna y las condiciones correspondientes
tabla_condiciones <- data.frame(
  sample = colnames(dat_merge_all_Kallisto),
  condition = c(
    "Male_distal", "Male_distal", "Male_distal", "Male_distal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Male_distal", "Male_distal",
    "Male_distal", "Male_distal", "Male_proximal", "Male_proximal", "Male_proximal",
    "Male_proximal", "Male_distal", "Male_distal", "Male_distal", "Male_distal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Male_proximal", "Male_distal",
    "Male_distal", "Male_distal", "Male_distal", "Male_proximal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_distal", "Male_distal", "Male_distal",
    "Male_distal", "Male_proximal", "Male_proximal", "Male_proximal", "Male_proximal",
    "Male_distal", "Male_distal", "Male_distal", "Male_distal", "Male_proximal",
    "Male_proximal", "Male_proximal", "Male_proximal", "Female_distal", "Female_distal",
    "Female_distal", "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_proximal", "Female_distal", "Female_distal", "Female_distal", "Female_distal",
    "Female_proximal", "Female_proximal", "Female_proximal", "Female_proximal", "Female_distal",
    "Female_distal", "Female_distal", "Female_distal", "Female_proximal", "Female_proximal",
    "Female_proximal", "Female_proximal", "Female_distal", "Female_distal", "Female_distal",
    "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_distal", "Female_distal", "Female_distal", "Female_distal", "Female_proximal",
    "Female_proximal", "Female_proximal", "Female_proximal", "Female_distal", "Female_distal",
    "Female_distal", "Female_distal", "Female_proximal", "Female_proximal", "Female_proximal",
    "Female_proximal"
  )
)


```

```{r}
colnames(dat_merge_all_Kallisto) <- tabla_condiciones$condition
```

```{r}
#col_order <- order(duplicated(colnames(matrix_dat_order)), colnames(matrix_dat_order))


col_order <- order(colnames(dat_merge_all_Kallisto), colnames(dat_merge_all_Kallisto), duplicated(colnames(dat_merge_all_Kallisto)))
```

```{r}
# Subsetting the data for significant genes in the specified gene family

subset_data_GPCRs<- dat_merge_all_Kallisto[significant_genes_GPCRs, col_order]

```

```{r}
# Plotting heatmap for the specific gene family
pheatmap(log2(subset_data_GPCRs + 1), cluster_cols = FALSE, cluster_rows = FALSE, scale = 'none', gaps_col = c(24, 48, 72, 96))
```

### EnhancedVolcano

```{r}
if (!requireNamespace('BiocManager', quietly = TRUE))
    install.packages('BiocManager')

  BiocManager::install('EnhancedVolcano')
```

```{r}
devtools::install_github('kevinblighe/EnhancedVolcano')

```

```{r}
library(EnhancedVolcano)

```

### Fucntion for Volcano plot

#### iGluRs/IRs

```{r}
generateEnhancedVolcano <- function(inputData, geneNames, titleText, subtitleText) {
  lab_IR <- paste0("italic('", rownames(inputData), "')")
  selectLab_italics <- paste0("italic('", geneNames, "')")

  result_plot <- EnhancedVolcano(inputData,
                  lab = lab_IR,
                  x = 'log2FoldChange',
                  y = 'padj',
                  title = titleText,  # Set the title from the function argument
                  subtitle = subtitleText,  # Set the subtitle from the function argument
                  selectLab = selectLab_italics,
                  xlab = bquote(~Log[2]~ 'fold change'),
                  pCutoff = 10e-32,
                  FCcutoff = 0.5,
                  pointSize = 1.0,
                  labSize = 4.0,
                  labCol = 'black',
                  labFace = 'bold',
                  boxedLabels = FALSE,
                  parseLabels = TRUE,
                  legendLabels = c("NS", expression(Log[2] ~ FC), "padj", expression(padj ~ and ~ log[2] ~ FC)),
                  col = c('black', '#B28B7A', '#E9CA68', 'red'),
                  colAlpha = 1,
                  legendPosition = "top",
                  legendLabSize = 14,
                  legendIconSize = 2.0,
                  legendDropLevels = TRUE,
                  drawConnectors = TRUE,
                  widthConnectors = 0.3,
                  colConnectors = 'black',
                  max.overlaps = 25,
                  ylim = c(0, 300),
                  xlim = c(-5.5, 10)) + theme_classic()
  #coord_flip()
  
  return(result_plot)
}


```

```{r}
# Example usage:
# solo plotie los down!
result_plot_TratmentA <- generateEnhancedVolcano(resKallisto_Treatmenta, c('STRG.7561.1.p2', 'STRG.5450.2.p3', 'STRG.5450.2.p2', 'STRG.26518.1.p1', 'STRG.26520.1.p4', 'STRG.4504.2.p1', 'STRG.5450.2.p1', 'STRG.26540.1.p1', 'STRG.45426.1.p1', 'STRG.19212.2.p1', 'STRG.26541.2.p1', 'STRG.49563.1.p1', 'STRG.49562.1.p1', 'STRG.56703.1.p1'

), "DESeq2 results Differential expression - Male distal vs Male proximal", "iGluRs/IRs")

```

#### TA. Male distal vs Male proximal - iGluRs/IRs

```{r}
result_plot_TratmentA
```

```{r}
result_plot_TratmentD <- generateEnhancedVolcano(resKallisto_Treatmentd, c('STRG.50985.1.p1', 'STRG.50989.1.p2', 'STRG.5450.2.p2', 'STRG.33530.1.p1', 'STRG.7561.1.p2', 'STRG.50993.1.p1', 'STRG.50992.1.p1', 'STRG.33519.1.p3', 'STRG.33350.1.p1', 'STRG.52918.5.p1', 'STRG.5450.2.p3', 'STRG.26540.1.p1', 'STRG.1974.1.p1', 'STRG.33350.2.p1', 'STRG.5450.2.p1', 'STRG.2001.1.p1', 'STRG.26541.2.p1', 'STRG.19212.2.p1', 'STRG.14809.2.p1', 'STRG.18882.2.p1', 'STRG.35168.1.p1', 'STRG.26541.3.p2'), 
                                                 
"DESeq2 results Differential expression - Male proximal vs Female proximal", "iGluRs/IRs" )
```

#### TD.DESeq2 results Differential expression - Male proximal vs Female proximal

```{r}
result_plot_TratmentD
```

#### TB.DESeq2 results Differential expression - Female_distal vs Female proximal

```{r}
result_plot_TratmentB <- generateEnhancedVolcano(resKallisto_Treatmentb, c(
'STRG.50985.1.p1', 'STRG.7561.1.p2', 'STRG.5450.2.p3', 'STRG.5450.2.p2', 'STRG.50993.1.p1', 'STRG.26520.1.p4', 'STRG.52918.5.p1', 'STRG.26518.1.p1', 'STRG.4504.2.p1', 'STRG.50989.1.p2', 'STRG.50992.1.p1', 'STRG.11765.1.p1', 'STRG.5450.2.p1', 'STRG.38406.1.p1', 'STRG.2159.1.p1', 'STRG.39111.1.p2', 'STRG.26540.1.p1', 'STRG.46044.3.p1', 'STRG.45427.1.p1', 'STRG.26541.2.p1', 'STRG.45426.1.p1', 'STRG.33530.1.p1', 'STRG.15456.1.p1', 'STRG.26541.3.p2', 'STRG.21210.1.p1', 'STRG.50071.1.p1', 'STRG.33523.1.p1', 'STRG.33525.2.p1', 'STRG.46044.1.p1', 'STRG.46275.1.p2'), "DESeq2 results Differential expression - Female distal vs Female proximal", "iGluRs/IRs")
```

```{r}
result_plot_TratmentB
```

#### TC. DESeq2 results Differential expression - Male_distal vs Female_distal, "GR/GPCRs"

```{r}
result_plot_TratmentC <- generateEnhancedVolcano(resKallisto_Treatmentc, c('STRG.50985.1.p1', 'STRG.50989.1.p2', 'STRG.5450.2.p2', 'STRG.33530.1.p1', 'STRG.7561.1.p2', 'STRG.50993.1.p1', 'STRG.50992.1.p1', 'STRG.33519.1.p3', 'STRG.33350.1.p1', 'STRG.52918.5.p1', 'STRG.5450.2.p3', 'STRG.26540.1.p1', 'STRG.1588.1.p1', 'STRG.1974.1.p1', 'STRG.11600.1.p2', 'STRG.33350.2.p1', 'STRG.5450.2.p1', 'STRG.11767.1.p1', 'STRG.2001.1.p1', 'STRG.26541.2.p1', 'STRG.19212.2.p1', 'STRG.14809.2.p1', 'STRG.18882.2.p1', 'STRG.4504.2.p1', 'STRG.35168.1.p1', 'STRG.26541.3.p2', 'STRG.20575.1.p7', 'STRG.33530.2.p2', 'STRG.62762.1.p1', 'STRG.45427.1.p1', 'STRG.46031.3.p1', 'STRG.29634.2.p1', 'STRG.2159.1.p1', 'STRG.43176.1.p1', 'STRG.21210.1.p1', 'STRG.46275.1.p2', 'STRG.39111.1.p2', 'STRG.26560.1.p3', 'STRG.7819.1.p1', 'STRG.26560.2.p3', 'STRG.11770.2.p1', 'STRG.33519.2.p1', 'STRG.29632.1.p1', 'STRG.61662.1.p1'
), "DESeq2 results Differential expression - Male distal vs Female distal", "iGluRs/IRs")
```

```{r}
result_plot_TratmentC
```

#### TA. Male distal vs Male proximal GR/GPCRs

```{r}
# Example usage:
# solo plotie los down!
result_plot_TratmentA_GR <- generateEnhancedVolcano(resKallisto_Treatmenta,
                                                    c('STRG.56530.1.p1', 'STRG.45731.1.p1', 'STRG.25665.1.p2', 'STRG.218.1.p1', 'STRG.25665.1.p1', 'STRG.19057.1.p1', 'STRG.10100.2.p1', 'STRG.11903.1.p2', 'STRG.29402.1.p1', 'STRG.52153.1.p4', 'STRG.7817.1.p1', 'STRG.52154.1.p1', 'STRG.21837.1.p1', 'STRG.32549.1.p1', 'STRG.45228.1.p1', 'STRG.36825.1.p1', 'STRG.27445.1.p2', 'STRG.21836.1.p1', 'STRG.32700.1.p2', 'STRG.13962.1.p1', 'STRG.50419.1.p1'
), "DESeq2 results Differential expression - Male distal vs Male proximal", "GRs/GPCRs")

```

```{r}
result_plot_TratmentA_GR
```

#### TB. Female_distal vs Female proximal GR/GPCRs

```{r}
# Example usage:
# solo plotie los down!
result_plot_TratmentB_GR <- generateEnhancedVolcano(resKallisto_Treatmenta,
                                                    c('STRG.56530.1.p1', 'STRG.21837.1.p1', 'STRG.45731.1.p1', 'STRG.11903.1.p2', 'STRG.218.1.p1', 'STRG.22259.1.p2', 'STRG.52154.1.p1', 'STRG.21836.1.p1', 'STRG.35198.1.p2', 'STRG.22577.1.p3', 'STRG.19037.1.p2', 'STRG.10100.2.p1', 'STRG.32549.1.p1'
), "DESeq2 results Differential expression - Female distal vs Female proximal", "GRs/GPCRs")

```

```{r}
result_plot_TratmentB_GR 
```

#### TC. Male_distal vs Female_distal GR/GPCRs

```{r}
# Example usage:
# solo plotie los down!
result_plot_TratmentC_GR <- generateEnhancedVolcano(resKallisto_Treatmenta,
                                                    c('STRG.45731.1.p1', 'STRG.68983.1.p1', 'STRG.56530.1.p1', 'STRG.44966.1.p1', 'STRG.10100.2.p1', 'STRG.65723.1.p1', 'STRG.25665.1.p2', 'STRG.52154.1.p1', 'STRG.19057.1.p1', 'STRG.54336.1.p1', 'STRG.24800.2.p1', 'STRG.52153.1.p4', 'STRG.218.1.p1', 'STRG.22259.1.p2', 'STRG.7817.1.p1', 'STRG.30251.1.p1', 'STRG.25665.1.p1', 'STRG.36825.1.p1', 'STRG.22259.3.p4', 'STRG.26574.1.p2', 'STRG.50770.1.p1', 'STRG.34226.3.p1', 'STRG.29402.1.p1', 'STRG.35198.1.p2', 'STRG.11903.2.p1', 'STRG.33106.1.p1', 'STRG.45228.1.p1', 'STRG.24531.1.p2', 'STRG.35285.1.p1', 'STRG.11903.1.p2', 'STRG.19037.1.p2', 'STRG.27445.1.p2', 'STRG.63515.1.p1', 'STRG.30251.2.p1'
), "DESeq2 results Differential expression - Male_distal vs Female_distal", "GRs/GPCRs")

```

```{r}
result_plot_TratmentC_GR
```

#### TD.DESeq2 results Differential expression - Male proximal vs Female proximal

```{r}
# Example usage:
# solo plotie los down!
result_plot_TratmentD_GR <- generateEnhancedVolcano(resKallisto_Treatmenta,
                                                    c('STRG.45731.1.p1', 'STRG.56530.1.p1', 'STRG.44966.1.p1', 'STRG.10100.2.p1', 'STRG.65723.1.p1', 'STRG.25665.1.p2', 'STRG.52154.1.p1', 'STRG.19057.1.p1', 'STRG.52153.1.p4', 'STRG.218.1.p1', 'STRG.22259.1.p2', 'STRG.7817.1.p1', 'STRG.30251.1.p1', 'STRG.25665.1.p1', 'STRG.36825.1.p1', 'STRG.22259.3.p4', 'STRG.26574.1.p2', 'STRG.50770.1.p1', 'STRG.34226.3.p1', 'STRG.29402.1.p1', 'STRG.11903.2.p1', 'STRG.33106.1.p1', 'STRG.45228.1.p1', 'STRG.24531.1.p2', 'STRG.11903.1.p2', 'STRG.27445.1.p2', 'STRG.30251.2.p1'
), "DESeq2 results Differential expression - Male proximal vs Female proximal", "GRs/GPCRs")

```

```{r}
result_plot_TratmentD_GR
```

#### TA. Barplots

```{r}
# Required Libraries
library(ggplot2)
library(dplyr)

# Provided data
data <- data.frame(
  Category = c("Total", "Upregulated", "Downregulated", "PANTHER", "PFAM", "InterPro", "Significant", "Blast", "CDS_Complete",
               "Total", "Upregulated", "Downregulated", "PANTHER", "PFAM", "InterPro", "Significant", "Blast", "CDS_Complete"),
  Total = c(111, 75, 36, 78, 86, 26, 40, 99, 71, 111, 88, 23, 34, 40, 37, 21, 24, 87),
  Receptor = c("iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs",
               "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs")
)

# Define the order of categories
desired_order <- c("Total", "Upregulated", "Downregulated", "PANTHER", "PFAM", "InterPro", "Significant", "Blast", "CDS_Complete")

# Grouped data by Receptor and Category, calculating the sums
grouped_data <- data %>%
  group_by(Receptor, Category) %>%
  summarise(Total = sum(Total), .groups = "drop") %>%
  mutate(Category = factor(Category, levels = desired_order))  # Setting the factor levels

# Define custom colors for the bars
custom_colors <- c('darkgrey', 'black')

# Plotting the bar plot with custom colors
plot <- ggplot(grouped_data, aes(x = Category, y = Total, fill = Receptor)) +
  geom_col(stat = "identity", position = "dodge", width = 0.5) +
  geom_text(aes(label = Total), vjust = -0.5, position = position_dodge(width = 0.8)) +
  labs(title = "", x = "Category", y = "Total") +
  scale_fill_manual(values = custom_colors) +  # Setting custom colors for the bars
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+scale_y_continuous(expand = c(0,0)) 

# Show the plot
p1<-print(plot)

```

#### TB. Barplots

```{r}
# Required Libraries
library(ggplot2)
library(dplyr)

# Provided data (updated)
data <- data.frame(
  Category = c("Total", "Upregulated", "Downregulated", "PANTHER", "PFAM", "InterPro", "Significant", "Blast", "CDS_Complete",
               "Total", "Upregulated", "Downregulated", "PANTHER", "PFAM", "InterPro", "Significant", "Blast", "CDS_Complete"),
  Total = c(111, 81, 30, 78, 85, 23, 30, 99, 71, 111, 89, 22, 34, 40, 37, 13, 24, 87),
  Receptor = c("iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs",
               "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs")
)

# Define the order of categories
desired_order <- c("Total", "Upregulated", "Downregulated", "PANTHER", "PFAM", "InterPro", "Significant", "Blast", "CDS_Complete")

# Grouped data by Receptor and Category, calculating the sums
grouped_data <- data %>%
  group_by(Receptor, Category) %>%
  summarise(Total = sum(Total), .groups = "drop") %>%
  mutate(Category = factor(Category, levels = desired_order))  # Setting the factor levels

# Define custom colors for the bars
custom_colors <- c('darkgrey', 'black')

# Plotting the bar plot with custom colors
plot <- ggplot(grouped_data, aes(x = Category, y = Total, fill = Receptor)) +
  geom_bar(stat = "identity", position = "dodge", width = 0.5) +
  geom_text(aes(label = Total), vjust = -0.5, position = position_dodge(width = 0.8)) +
  labs(title = "", x = "Category", y = "Total") +
  scale_fill_manual(values = custom_colors) +  # Setting custom colors for the bars
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Show the plot
pb<-print(plot)

```

#### TC. Barplot

```{r}
# Required Libraries
library(ggplot2)
library(dplyr)

# Provided data (updated)
data <- data.frame(
  Category = c("Total", "Upregulated", "Downregulated", "PANTHER", "PFAM", "InterPro", "Significant", "Blast", "CDS_Complete",
               "Total", "Upregulated", "Downregulated", "PANTHER", "PFAM", "InterPro", "Significant", "Blast", "CDS_Complete"),
  Total = c(111, 88, 23, 78, 85, 23, 44, 99, 77, 111, 82, 29, 34, 40, 37, 34, 20, 87),
  Receptor = c("iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs",
               "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs")
)

# Define the order of categories
desired_order <- c("Total", "Upregulated", "Downregulated", "PANTHER", "PFAM", "InterPro", "Significant", "Blast", "CDS_Complete")

# Grouped data by Receptor and Category, calculating the sums
grouped_data <- data %>%
  group_by(Receptor, Category) %>%
  summarise(Total = sum(Total), .groups = "drop") %>%
  mutate(Category = factor(Category, levels = desired_order))  # Setting the factor levels

# Define custom colors for the bars
custom_colors <- c('darkgrey', 'black')

# Plotting the bar plot with custom colors
plot <- ggplot(grouped_data, aes(x = Category, y = Total, fill = Receptor)) +
  geom_bar(stat = "identity", position = "dodge", width = 0.5) +
  geom_text(aes(label = Total), vjust = -0.5, position = position_dodge(width = 0.8)) +
  labs(title = "", x = "Category", y = "Total") +
  scale_fill_manual(values = custom_colors) +  # Setting custom colors for the bars
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Show the plot
pb<-print(plot)

```

#### TD. Barplot

```{r}
# Required Libraries
library(ggplot2)
library(dplyr)

# Provided data (updated)
data <- data.frame(
  Category = c("Total", "Upregulated", "Downregulated", "PANTHER", "PFAM", "InterPro", "Significant", "Blast", "CDS_Complete",
               "Total", "Upregulated", "Downregulated", "PANTHER", "PFAM", "InterPro", "Significant", "Blast", "CDS_Complete"),
  Total = c(111, 87, 23, 78, 85, 23, 44, 99, 71, 111, 82, 29, 34, 40, 37, 34, 24, 87),
  Receptor = c("iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs", "iGluRs_Irs",
               "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs", "GPCRs_GRs")
)

# Define the order of categories
desired_order <- c("Total", "Upregulated", "Downregulated", "PANTHER", "PFAM", "InterPro", "Significant", "Blast", "CDS_Complete")

# Grouped data by Receptor and Category, calculating the sums
grouped_data <- data %>%
  group_by(Receptor, Category) %>%
  summarise(Total = sum(Total), .groups = "drop") %>%
  mutate(Category = factor(Category, levels = desired_order))  # Setting the factor levels

# Define custom colors for the bars
custom_colors <- c('darkgrey', 'black')

# Plotting the bar plot with custom colors
plot <- ggplot(grouped_data, aes(x = Category, y = Total, fill = Receptor)) +
  geom_bar(stat = "identity", position = "dodge", width = 0.5) +
  geom_text(aes(label = Total), vjust = -0.5, position = position_dodge(width = 0.8)) +
  labs(title = "", x = "Category", y = "Total") +
  scale_fill_manual(values = custom_colors) +  # Setting custom colors for the bars
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Show the plot
print(plot)

```



















